File: rubyonrails/notejam/app/controllers/pad_controller.rb
class PadController < ApplicationController
  before_action :authenticate_user
  def create
    if params[:pad]
      @pad = Pad.new(pad_params.merge(user: current_user))
      if @pad.save
        redirect_to url_for(:all_notes), flash: { success: 'Pad is successfully created' }
      end
    end
  end

  def edit
    @pad = current_user.pads.find(params[:id])
    @current_name = @pad.name
    if params[:pad]
      if @pad.update(pad_params)
        redirect_to view_pad_notes_path(id: @pad.id), flash: { success: 'Pad is successfully updated' }
      end
    end
  end

  def delete
    @pad = current_user.pads.find(params[:id])
    if request.post?
      @pad.destroy
      redirect_to all_notes_path, flash: { success: 'Pad is successfully deleted' }
    end
  end

  def view
    @pad = current_user.pads.find(params[:id])
    @notes = @pad.notes.order(order_param)
  end

  private

  def pad_params
    params.require(:pad).permit(:name)
  end
end

File: laravel/notejam/app/controllers/PadController.php
<?php

class PadController extends BaseController {

    public function create()
    {
        if (Request::isMethod('post'))
        {
            $validation = $this->validator();
            if ($validation->fails())
            {
                return Redirect::route('create_pad')->withErrors($validation);
            }
            
            $pad = new Pad(array(
                'name' => Input::get('name')
            ));
            Auth::user()->pads()->save($pad);
            
            return Redirect::route('view_pad', array('id' => $pad->id))
                ->with('success', 'Pad is created.');
        }
        return View::make('pad/create');
    }

    public function edit($id)
    {
        $pad = $this->getPadOrFail($id);
        if (Request::isMethod('post'))
        {
            $validation = $this->validator();
            if ($validation->fails())
            {
                return Redirect::route('edit_pad', array('id' => $pad->id))
                    ->withErrors($validation);
            }
            
            $pad->name = Input::get('name');
            $pad->save();
            
            return Redirect::route('view_pad', array('id' => $pad->id))
                ->with('success', 'Pad is updated.');
        }
        return View::make('pad/edit', array('pad' => $pad));
    }

    public function view($id)
    {
        $pad = $this->getPadOrFail($id);
        $orderParams = $this->processOrderParam();
        
        $notes = $pad->notes()
            ->orderBy($orderParams[0], $orderParams[1])
            ->get();
        
        return View::make('pad/view', array(
            'pad' => $pad,
            'notes' => $notes
        ));
    }

    public function delete($id)
    {
        $pad = $this->getPadOrFail($id);
        if (Request::isMethod('post'))
        {
            $pad->delete();
            return Redirect::route('all_notes')
                ->with('success', 'Pad is deleted.');
        }
        
        return View::make('pad/delete', array('pad' => $pad));
    }

    private function getPadOrFail($id)
    {
        return Auth::user()->pads()
            ->where('id', '=', $id)
            ->firstOrFail();
    }

    private function validator()
    {
        return Validator::make(Input::all(), array(
            'name' => 'required',
        ));
    }

}

